pyfstat/tcw_fstat_map_funcs.py

```diff
@@ -122,6 +124,33 @@ def get_maxF_idx(self):
         """
         return np.unravel_index(np.argmax(self.F_mn), self.F_mn.shape)
 
+    def get_logBtSG(self, windowRange):
+        """Compute (natural log of the) transient-CW Bayes-factor B_tSG = P(x|HyptS)/P(x|HypG).
+
+        Here HypG = Gaussian noise hypothesis, HyptS = transient-CW signal hypothesis.
+
+        B_tSG is marginalized over start-time and timescale of transient CW signal,
+        using given type and parameters of transient window range.
+
+        This is a python port of the `lalpulsar.ComputeTransientBstat` implementation,
+        replacing for loops by numpy operations.
+        """
+        # step through F_mn array subtract maxF and sum e^{F_mn - maxF}
+        # The maximum-likelihood Fmax is globally subtracted from F_mn, and stored separatedly in the struct, because in most
+        # expressions it is numerically more robust to compute e^(F_mn - Fmax), which at worst can underflow, while
+        # e^F_mn can overflow (for F>~700). The constant offset e^Fmax is irrelevant for posteriors (normalization constant), or
+        # can be handled separately, eg by computing log(B) = Fmax + log(sum e^(Fmn-Fmax)) for the Bayes-factor.
+        sum_eB = np.sum(np.exp(-self.maxF - self.F_mn))
+        # combine this to final log(Bstat) result with proper normalization (assuming rhohMax=1):
+        logBhat = self.maxF + np.log(sum_eB)  # unnormalized Bhat
```

Comment:
So... No logsumexp on implementation but yes to test against?

Reply:
I've deliberately added a commit that introduces an extra function for it, and an immediate revert, so that we have this version in the history and can easily go back to testing against it if needed.